indexing-service/src/main/java/org/apache/druid/indexing/overlord/supervisor/SupervisorResource.java

```diff
@@ -560,6 +560,25 @@ public Response reset(@PathParam("id") final String id)
     return handleResetRequest(id, null);
   }
 
+  @POST
+  @Path("/{id}/setTaskCount")
+  @Produces(MediaType.APPLICATION_JSON)
+  @ResourceFilters(SupervisorResourceFilter.class)
+  public Response setTaskCount(@PathParam("id") final String id, @QueryParam("taskCount") final Integer taskCount)
```

Comment:
Instead of doing this via a new API that is scoped to just the `taskCount` field in the supervisor spec, can you update this API to be a PATCH request on `.../supervisors/{id}`

That would give us the flexibility to atomically update any field in the supervisor spec instead of needing a new API for each supervisor spec modification

Reply:
I agree, this would be better than what is currently being done in this PR.

But I am still not sure why we need a separate API at all.
Couldn't this be an optimization in the existing `specPost` API itself?

- Post a supervisor spec
- If a supervisor already exists for the given id, we compare the spec against its current value.
- If the change is only in task count, don't go via the normal route and invoke `updateTaskCount` method instead.
- Even if we use the PATCH API route, we would still need a similar diff mechanism.

cc: @adithyachakilam , @suneet-s

Reply:
We are trying to avoid an extra `GET` call to get the actual existing spec to send to `POST` call. Also, the idea was to try libraries like https://github.com/java-json-tools/json-patch to support editing of any fields dynamically. But a standard `PATCH` controller seems like not availbale.

Reply:
What's the harm in doing a GET before the POST? It is not a very expensive operation.
In fact, it is desirable to know the current state before we try to update it.

> and it could also change in between those calls.

This is always a possibility. Some other user could update the task count right after we updated it. Having a separate API does not protect against concurrent updates. Updating a supervisor spec is typically an admin action and is also audited. The likelihood of concurrent updates should be very low.

> Also, the idea was to try libraries like https://github.com/java-json-tools/json-patch to support editing of any fields dynamically.

Being able to update separate fields is a valid requirement, but the library we end up using should not dictate the contract of our APIs.

@adithyachakilam , Could you please share some more details on the challenges with the current API?

Reply:
> It is not a very expensive operation.

It does end up being expensive when we build automation solutions on top of it.


> Some other user could update the task count right after we updated it.

Sure, In that case two sequential updates are done. But when we do a `GET` and a `POST` a valid user update made in between might be overridden.


> but the library we end up using should not dictate the contract of our APIs.

It does follow of [RFC 6902 (JSON Patch)](http://tools.ietf.org/html/rfc6902) and [RFC 7386 (JSON Merge Patch)](http://tools.ietf.org/html/rfc7386) which are standard for patch requests merge specification.